go-controller/pkg/ovn/endpoints.go

```diff
@@ -57,69 +62,47 @@ func (ovn *Controller) AddEndpoints(ep *kapi.Endpoints) error {
 	}
 	tcpPortMap := make(map[string]lbEndpoints)
 	udpPortMap := make(map[string]lbEndpoints)
-	ovn.getLbEndpoints(ep, tcpPortMap, udpPortMap)
-	for svcPortName, lbEps := range tcpPortMap {
-		ips := lbEps.IPs
-		targetPort := lbEps.Port
-		for _, svcPort := range svc.Spec.Ports {
-			if svcPort.Protocol == kapi.ProtocolTCP && svcPort.Name == svcPortName {
-				if util.ServiceTypeHasNodePort(svc) {
-					klog.V(5).Infof("Creating Gateways IP for NodePort: %d, %v", svcPort.NodePort, ips)
-					err = ovn.createGatewaysVIP(svcPort.Protocol, svcPort.NodePort, targetPort, ips)
-					if err != nil {
-						klog.Errorf("Error in creating Node Port for svc %s, node port: %d - %v\n", svc.Name, svcPort.NodePort, err)
-						continue
-					}
-				}
-				if util.ServiceTypeHasClusterIP(svc) {
-					var loadBalancer string
-					loadBalancer, err = ovn.getLoadBalancer(svcPort.Protocol)
-					if err != nil {
-						klog.Errorf("Failed to get loadbalancer for %s (%v)",
-							svcPort.Protocol, err)
-						continue
-					}
-					err = ovn.createLoadBalancerVIP(loadBalancer,
-						svc.Spec.ClusterIP, svcPort.Port, ips, targetPort)
-					if err != nil {
-						klog.Errorf("Error in creating Cluster IP for svc %s, target port: %d - %v\n", svc.Name, targetPort, err)
-						continue
-					}
-					vip := util.JoinHostPortInt32(svc.Spec.ClusterIP, svcPort.Port)
-					ovn.AddServiceVIPToName(vip, svcPort.Protocol, svc.Namespace, svc.Name)
-					ovn.handleExternalIPs(svc, svcPort, ips, targetPort, false)
-				}
-			}
-		}
+	sctpPortMap := make(map[string]lbEndpoints)
+	ovn.getLbEndpoints(ep, tcpPortMap, udpPortMap, sctpPortMap)
+
+	protoPortMap := map[kapi.Protocol]map[string]lbEndpoints{
+		kapi.ProtocolTCP:  tcpPortMap,
+		kapi.ProtocolUDP:  udpPortMap,
+		kapi.ProtocolSCTP: sctpPortMap,
 	}
```

Comment:
just do

    func (ovn *Controller) getLbEndpoints(ep *kapi.Endpoints) map[kapi.Protocol]map[string]lbEndpoints {

Reply:
I'll modify the function to just return the map. It is awkwardly setting map values.